src/NServiceBus.RavenDB/Subscriptions/SubscriptionData.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace NServiceBus.Persistence.SubscriptionStorage
+{
+    using System;
+    using System.Security.Cryptography;
+    using System.Text;
+    using Unicast.Subscriptions;
+    using Unicast.Subscriptions.MessageDrivenSubscriptions;
+
+    class SubscriptionData
+    {
+        public string Id { get; set; }
+        public string TransportAddress { get; set; }
+        public string Endpoint { get; set; }
+        //TODO: do we need more info here, e.g. MessageType or is that sufficient?
+
+        public static string FormatId(MessageType messageType, Subscriber subscriber)
+        {
+            // use MD5 hash to get a 16-byte hash of the string
+            using (var provider = new MD5CryptoServiceProvider())
```

Comment:
Since this is new code we probably shouldn't get back in bed with MD5 and use SHA1CryptoServiceProvider instead.

Which would mean not using a Guid (too many bytes) and converting the bytes to hex instead.